CMakeLists.txt

```diff
@@ -45,6 +45,8 @@ if(NOT DISABLE_JEMALLOC)
     list(APPEND EXTERNAL_LIBS PRIVATE jemalloc)
 endif()
 
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "do not build shared libs by default")
```

Comment:
Why do we set this flag on root CMakeLists also?

BTW, the comment seems magic without background (why forbid shared lib?).

Reply:
I think we just want to build static libraries of dependences and link them statically, so I added it to CMakeLists to avoid some dependencies set this cache variable to true in their cmake to pollute this configuration. (cache variables can only be assigned once, and the next assignment needs to specify `FORCE`, otherwise it is invalid)

Reply:
Static libraries allow users to not worry about library versioning issues and linking issues, and are more portable (just like go / rust, statically link almost everything), which I think is probably secondary to the increase in size of these binaries.

Reply:
Thanks for your explanation. Sounds good to me.